server/src/tree-sitter/analyzer.ts

```diff
@@ -701,6 +702,64 @@ export default class Analyzer {
       }
     }
   }
+
+  public processRecipeScanResults (scanResults: string, uri: any): RequestResult['ProcessRecipeScanResults'] {
+    let uriString
+    if (typeof uri === 'string') {
+      uriString = uri
+    }
+
+    if (uri.fsPath !== undefined) { // vscode.Uri
+      uriString = uri.fsPath
+    }
+
+    if (typeof uriString !== 'string') {
+      logger.debug('[ProcessRecipeScanResults] Cannot obtain the file uri, abort processing scan results')
+      return undefined
+    }
+
+    if (!uriString.startsWith('file://')) {
+      uriString = 'file://' + uriString
+    }
+
+    if (this.uriToAnalyzedDocument[uriString] === undefined) {
+      logger.debug(`[ProcessRecipeScanResults] Analyzed document for ${uriString} not found, abort processing scan results`)
+      return undefined
+    }
+
+    const lines = scanResults.split('\n')
+    const index = lines.findIndex((line) => line.includes('INCLUDE HISTORY'))
+    if (index === -1) {
+      logger.debug('[Analyzer] Cannot find INCLUDE HISTORY in scan results, abort processing scan results')
+      return undefined
+    }
+
+    const validTexts = lines.slice(index).join('\n')
+    const scanResultdocumentUri = uriString + '.scanned'
+    const scanResultDocument = TextDocument.create(
+      scanResultdocumentUri,
```

Comment:
So we are copying the results again to a file? I think it would be more optimal to do this copy in the client and passing the filename rather than passing the whole content in the request. You could use/extract the logic from embedded language documents storage.

Reply:
This is removed in https://github.com/yoctoproject/vscode-bitbake/pull/73, which is the PR that applies the scan results to the document. You may review this part in that PR after this one is merged. Besides, the `TextDocument.create` won't create a real file. 
More details:
https://github.com/yoctoproject/vscode-bitbake/pull/73/files#diff-43679d42e59e68b6c9bac74708dbc60a53568a147bf342fa3e301f231e930b96R778